File: terraform/modules/content_delivery_network/lambda_functions/reset_response_header_caching/src/lambda_function.js
exports.handler = async (event, context) => {

    function has(object, key) {
      return object ? hasOwnProperty.call(object, key) : false;
   }

    const response = event.Records[0].cf.response;
    const headers = response.headers;

    if (typeof(headers.expires) !== 'undefined') {delete headers.expires;}

    headers['cache-control'] = [{
        key:   'Cache-Control',
        value: "max-age=21600"
    }];

    console.log("Overwrite headers for " + response.uri);
    return response;
};

File: terraform/modules/content_delivery_network/lambda_functions/response_header_cache_control/src/lambda_function.py
# mypy: ignore-errors


def handler(event, context):
    """Add a response header to keep the response in cache for 1 year"""
    response = event["Records"][0]["cf"]["response"]
    response["headers"]["cache-control"] = [
        {"key": "Cache-Control", "value": "max-age=31536000"}
    ]

    return response
